### ISSUE TITLE
Web app and CLI to create Comunica engine

### ISSUE THREAD

**Issue by pheyvaer** [comunica/comunica#646]
#### Issue type:

- :heavy_plus_sign: Feature request

____
#### Description:

Comunica engines can have a lot of features, such as support for SPARQL endpoints, local files, remote files, JSON-LD, HDT, and GraphQL-LD. Building a custom engine that only contains the desired features is not straight forward for new users and is repetitive even for users who have experience with the code base. A solution could be a CLI and Web app that creates a ready-to-use engine based on a set of parameters provided by a user. See it as `npm init` but for Comunica engines.

**Comment by pheyvaer** [comment]
Regarding effort, providing the CLI is not difficult, but having a library that takes the parameters from the CLI and turns it into an engine is from my point.

**Comment by rubensworks** [comment]
Basically, this tool would have to output a (basic) npm package, containing 1) a Comunica config, and 2) a package.json containing all the proper dependencies.

For reference, this should replace the current [Comunica packager](https://github.com/comunica/comunica/tree/master/packages/packager). It could serve as a starting point for this tool though.

> Regarding effort, providing the CLI is not difficult, but having a library that takes the parameters from the CLI and turns it into an engine is, from my point.

Indeed, depending on how far we want to go with this, this would also be able to detect actor parameters, and allow those to be configured. But for a first version, this is probably not needed.

**Comment by pheyvaer** [comment]
> Basically, this tool would have to output a (basic) npm package, containing 1) a Comunica config, and 2) a package.json containing all the proper dependencies.

Indeed. You can build it, but not have to understand. And you can adjust afterwards if needed.

> For reference, this should replace the current Comunica packager. It could serve as a starting point for this tool though.

Does it? What if I update the config don't I need to package it again?

**Comment by rubensworks** [comment]
> > For reference, this should replace the current Comunica packager. It could serve as a starting point for this tool though.

> Does it? What if I update the config don't I need to package it again?

Good point. Perhaps the tool should also be able to handle this?

**Comment by pheyvaer** [comment]
> Good point. Perhaps the tool should also be able to handle this?

That is solution. So you could have `comunica-engine init` and `comunica-engine package`, similar to `npm init` and `npm install`. What do you think?

**Comment by rubensworks** [comment]
> So you could have comunica-engine init and comunica-engine package, similar to npm init and npm install. What do you think?

The command name may be a bit too generic.
Now that I think of it, `comunica-package` actually seems like a good name to me.
Perhaps it might make sense to just implement all of this into the current packager (full rewrite).

`init` sub-command sounds good.
I would replace `package` sub-command with something more specific, like `reload-dependencies`.

**Comment by pheyvaer** [comment]
Sure sure. The command was just for my example :smile: But your suggestions sound good to me :+1:

**Comment by rubensworks** [comment]
Some inspiration: https://prismjs.com/download.html

**Comment by rubensworks** [comment]
Another nice extra feature: importing an existing config so that it can easily be modified.
